presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -209,36 +176,57 @@ public static Slice stPoint(@SqlType(DOUBLE) double x, @SqlType(DOUBLE) double y
     @SqlType(GEOMETRY_TYPE_NAME)
     public static Slice stMultiPoint(@SqlType("array(" + GEOMETRY_TYPE_NAME + ")") Block input)
     {
-        MultiPoint multipoint = new MultiPoint();
+        CoordinateSequence coordinates = readPointCoordinates(input, "ST_MultiPoint", false);
+        if (coordinates.size() == 0) {
+            return null;
+        }
+
+        return serialize(createJtsMultiPoint(coordinates));
+    }
+
+    private static CoordinateSequence readPointCoordinates(Block input, String functionName, boolean forbidDuplicates)
+    {
+        PackedCoordinateSequenceFactory coordinateSequenceFactory = new PackedCoordinateSequenceFactory();
+        double[] coordinates = new double[2 * input.getPositionCount()];
+        double lastX = Double.NaN;
+        double lastY = Double.NaN;
         for (int i = 0; i < input.getPositionCount(); i++) {
             if (input.isNull(i)) {
-                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Invalid input to ST_MultiPoint: null at index %s", i + 1));
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Invalid input to %s: null at index %s", functionName, i + 1));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;+&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 195. Consider formatting this data with &#39;%d&#39; instead.